Unithereum.CodeGen/UnithereumCodeGenProcessor.cs

```diff
@@ -15,84 +12,73 @@ public class UnithereumCodeGenProcessor : AssetPostprocessor
     {
         public override int GetPostprocessOrder() => 0;
 
-        private static readonly HashSet<string> CsharpKeywords = new HashSet<string>
-        {
-            "abstract", "as", "base", "bool", "break", "byte", "case", "catch", "char", "checked", "class", "const",
-            "continue", "decimal", "default", "delegate", "do", "double", "else", "enum", "event", "explicit", "extern",
-            "false", "finally", "fixed", "float", "for", "foreach", "goto", "if", "implicit", "in", "int", "interface",
-            "internal", "is", "lock", "long", "namespace", "new", "null", "object", "operator", "out", "override",
-            "params", "private", "protected", "public", "readonly", "ref", "return", "sbyte", "sealed", "short",
-            "sizeof", "stackalloc", "static", "string", "struct", "switch", "this", "throw", "true", "try", "typeof",
-            "uint", "ulong", "unchecked", "unsafe", "ushort", "using", "virtual", "void", "volatile", "while"
-        };
-
         /// <summary>
         /// Detect .abi and .bin files on import and generate code.
         /// </summary>
         public void OnPreprocessAsset()
         {
             var changedPath = Path.Combine(Path.GetDirectoryName(Application.dataPath)!, assetPath);
-            if (changedPath.EndsWith(".abi", StringComparison.OrdinalIgnoreCase))
-            {
-                var binPath = Path.ChangeExtension(changedPath, ".bin");
-                Generate(changedPath, File.Exists(binPath) ? binPath : null);
-            }
-            else if (changedPath.EndsWith(".bin", StringComparison.OrdinalIgnoreCase))
-            {
-                var abiPath = Path.ChangeExtension(changedPath, ".abi");
-                if (File.Exists(abiPath)) Generate(abiPath, changedPath);
-            }
+            FindAbiBinPairAndGenerate(changedPath);
         }
 
         /// <summary>
-        /// Search for all .abi and .bin files in the Asset/ directory and force import them.
+        ///     Search for all .abi and .bin files in the Asset/ directory and force import them.
         /// </summary>
-        /// <remarks>As the <see cref="OnPreprocessAsset"/> function will generate code for imported .abi files,
-        /// this function will regenerate code for all .abi files without explicitly calling <see cref="Generate"/>.
-        /// </remarks>
         [MenuItem("Unithereum/Regenerate All...")]
         private static void RegenerateAllMenu()
         {
+            var codeGenPath = Path.Combine(Application.dataPath, Config.GetConfig().OutputDir);
+            if (Directory.Exists(codeGenPath)) Directory.Delete(codeGenPath, true);
```

Comment:
Written in `README.md` that `outputDir` should be relative path to `Assets/` dir, would it be better to throw in `Config` constructor if given path is absolute?

And my mistake, I'll change it to delete after dialog.

Reply:
I think it might be beneficial to sort out absolute cases, especially because it doesn't make a whole lot of sense on Windows if they wrote something like "C:\foo" in config, since there cannot be a colon in windows path. Also, it might be ambiguous if "/foo/" means really "/asdf/" or "UnityProject/foo", so it might be useful for avoiding mistakes.